backend/api/Areas/Property/Mapping/Search/PropertyMap.cs

```diff
@@ -9,19 +9,41 @@ public class PropertyMap : IRegister
         public void Register(TypeAdapterConfig config)
         {
             config.NewConfig<Entity.Property, Model.PropertyModel>()
-                .Map(dest => dest.PropertyType, src => src.PropertyTypeId)
-                .Map(dest => dest.Classification, src => src.ClassificationId)
+                .Map(dest => dest.Id, src => src.Id)
+                .Map(dest => dest.RowVersion, src => src.RowVersion)
+                .Map(dest => dest.PID, src => src.ParcelIdentity)
+                .Map(dest => dest.PIN, src => src.PIN)
+                .Map(dest => dest.PropertyTypeId, src => src.PropertyTypeId)
+                .Map(dest => dest.PropertyType, src => src.PropertyType.Description)
```

Comment:
Can you confirm that none of these dereferences can result in an NPE? IE, are any of these relationships nullable.

Reply:
Not sure. I copied this from the other mapping from Jeremy's work. How do we check for nullable?

Reply:
src.PropertyType != null ? src.PropertyType.Description : null
or write an extension for the property entity, something like getPropertyDescription(), which can then just do src.PropertyType?.Description.

Reply:
happy to make that change but two questions:
- then what do we do with the other mapping that Jeremy created `Mapping/Property/PropertyMap.cs` that one has the same "faulty" code and it is used in the property info slideout
- How do we determine what is a required field (ie not null on DB) on the entities? 

I ask because by looking at Doug sql file, I see two things:
1. on PIMS_PROPERTY table "PROPERTY_TYPE_CODE"                 	nvarchar(20) NOT NULL," so that tells me prop type code cannot be null (this is mapped in EF to PropertyTypeId)
2. When I search for the FK constraint for the Property Type ("PIM_PRPTYP_PIM_PRPRTY_FK") I see that:
```sql
ALTER TABLE "dbo"."PIMS_PROPERTY"
	ADD CONSTRAINT "PIM_PRPTYP_PIM_PRPRTY_FK"
	FOREIGN KEY("PROPERTY_TYPE_CODE")
	REFERENCES "dbo"."PIMS_PROPERTY_TYPE"("PROPERTY_TYPE_CODE")
...
```
What I understand for this is that the FK will never be null because it uses the type code which cannot be null on PIMS_PROPERTY. What could I be missing here? 
Again, just trying to understand how we determine when things can be nullable in EF

Reply:
As long as the foreign key is not nullable, and references a non-nullable pk the null checks are not necessary. you can either check that in the ef fluent configuration or in the actual database itself. Would recommend not looking at Doug's scripts as they don't necessarily reflect the actual database.

Reply:
I looked at the latest EF designer for the current migration **20210909194500_v0.2.10.0.Designer.cs** and this is what I see:

```csharp
            modelBuilder.Entity("Pims.Dal.Entities.Property", b =>
                {
                    b.HasOne("Pims.Dal.Entities.Address", "Address")
                        .WithMany("Properties")
                        .HasForeignKey("AddressId")
                        .HasConstraintName("PIM_ADDRSS_PIM_PRPRTY_FK")
                        .OnDelete(DeleteBehavior.Restrict)
                        .IsRequired();

                    b.HasOne("Pims.Dal.Entities.PropertyAreaUnitType", "AreaUnit")
                        .WithMany("Properties")
                        .HasForeignKey("AreaUnitId")
                        .HasConstraintName("PIM_ARUNIT_PIM_PRPRTY_FK")
                        .OnDelete(DeleteBehavior.Restrict)
                        .IsRequired();

                    b.HasOne("Pims.Dal.Entities.PropertyClassificationType", "Classification")
                        .WithMany("Properties")
                        .HasForeignKey("ClassificationId")
                        .HasConstraintName("PIM_PRPCLT_PIM_PRPRTY_FK")
                        .OnDelete(DeleteBehavior.Restrict)
                        .IsRequired();

                    b.HasOne("Pims.Dal.Entities.PropertyDataSourceType", "DataSource")
                        .WithMany("Properties")
                        .HasForeignKey("DataSourceId")
                        .HasConstraintName("PIM_PIDSRT_PIM_PRPRTY_FK")
                        .OnDelete(DeleteBehavior.Restrict)
                        .IsRequired();

                    b.HasOne("Pims.Dal.Entities.District", "District")
                        .WithMany("Properties")
                        .HasForeignKey("DistrictId")
                        .HasConstraintName("PIM_DSTRCT_PIM_PRPRTY_FK")
                        .OnDelete(DeleteBehavior.Restrict)
                        .IsRequired();

                    b.HasOne("Pims.Dal.Entities.PropertyType", "PropertyType")
                        .WithMany("Properties")
                        .HasForeignKey("PropertyTypeId")
                        .HasConstraintName("PIM_PRPTYP_PIM_PRPRTY_FK")
                        .OnDelete(DeleteBehavior.Restrict)
                        .IsRequired();

                    b.HasOne("Pims.Dal.Entities.Region", "Region")
                        .WithMany("Properties")
                        .HasForeignKey("RegionId")
                        .HasConstraintName("PIM_REGION_PIM_PRPRTY_FK")
                        .OnDelete(DeleteBehavior.Restrict)
                        .IsRequired();

                    b.HasOne("Pims.Dal.Entities.PropertyStatusType", "Status")
                        .WithMany("Properties")
                        .HasForeignKey("StatusId")
                        .HasConstraintName("PIM_PRPSTS_PIM_PRPRTY_FK")
                        .OnDelete(DeleteBehavior.Restrict)
                        .IsRequired();

                    b.HasOne("Pims.Dal.Entities.PropertyTenureType", "Tenure")
                        .WithMany("Properties")
                        .HasForeignKey("TenureId")
                        .HasConstraintName("PIM_PRPTNR_PIM_PRPRTY_FK")
                        .OnDelete(DeleteBehavior.Restrict)
                        .IsRequired();
```

I think then there is no need for the null checks here as the FK constrains are marked required - right?